src/TrackWrapper.js

```diff
@@ -106,52 +110,6 @@ export default function TrackWrapper(props) {
 
     const transformedRowInfo = (!selectedRows ? rowInfo : selectedRows.map(i => rowInfo[i]));
 
-    /* TODO: Move the code below to a better place when we decide how to properly show DataTable */ 
-    // States and functions related to DataTable.
-    const [dataTableRows, setDataTableRows] = useState([]);
-    const [dataTableColumns, setDataTableColumns] = useState([]);
-    const [APIRequestStatus, setAPIRequestStatus] = useState({ msg: "", isLoading: false });
-    
-    function requestIntervalTfs(url) {
-        setAPIRequestStatus({msg: "Receiving Cistrome DB API response...", isLoading: true});
-
-        requestJSON(url, (error, data) => {
-            const keys = Object.keys(data);
-
-            // Not accesible or there is no result.
-            if(error !== null || keys.length === 0) {
-                setAPIRequestStatus({msg: "No TFs are found in this interval", isLoading: false});
-                setDataTableRows([]);
-                setDataTableColumns([]);
-                return;
```

Comment:
Would removing `return;` and moving the below code in an additional `else { ... }` statement resolve the problem as well? I think your change looks great but just wanna know better about the error.

(It is weird that I do not see the error msg in my browsers...)

Reply:
No I do not think that return statement is the issue. It is this return statement https://github.com/hms-dbmi/cistrome-higlass-wrapper/pull/163/files/a0201f68135c6a4cda987f3a253069803d948309#diff-adae11a6acfedf9f91b7f703a7c80679R62 that causes the issue.

If that return statement is executed, then these 3 calls to `useState()` are never reached https://github.com/hms-dbmi/cistrome-higlass-wrapper/compare/sehilyi/fix-137...keller-mark/table-refactoring?expand=1#diff-adae11a6acfedf9f91b7f703a7c80679L111

I think this is also why the error message does not always show up, since React may not detect the issue unless there are differences in the number of `useState()` calls from different renders. This docs page has some more info as well https://reactjs.org/docs/hooks-rules.html

Reply:
Oh, I got it. That might be the problem why I cannot see the error. Thanks for the detailed explanation!